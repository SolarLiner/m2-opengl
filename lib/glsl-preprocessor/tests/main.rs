use std::path::PathBuf;

use glsl_preprocessor::process_file;

const TESTS_FILES: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/tests/fixtures");

#[test]
fn test_process_file() {
    let tests_files = PathBuf::from(TESTS_FILES);
    let unwrapped = process_file(tests_files.join("shader.glsl"));
    insta::assert_snapshot!(unwrapped);
}