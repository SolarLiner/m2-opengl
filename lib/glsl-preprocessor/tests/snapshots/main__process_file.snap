---
source: lib/glsl-preprocessor/tests/main.rs
expression: unwrapped
---
#version 330 core

vec3 gen_color() {
    return vec3(1,0,1);
}


out vec4 out_color;

void main() {
    out_color = vec4(gen_color(), 1);
}

